include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -631,6 +631,12 @@ namespace VectorTools
         if (!cell1->active() && !cell2->active())
           continue;
 
+        // skip foreign cells
+        if (cell1->active() && !cell1->is_locally_owned())
+          continue;
+        if (cell2->active() && !cell2->is_locally_owned())
+          continue;
```

Comment:
But shouldn't you have the property that if one of the cells is locally owned then the other one needs to be as well? I think that should be possible to test.